### PR TITLE
introduce PHPStan (retry)

### PR DESCRIPTION
**Pull request by sasezaki**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

### Description
* At this PR set up target level as 5, because level 6 is a lot of "no value type specified" occurred. and I feel requires remove current deprecations before level up to 6 or over.
* travis setup is borrowed from https://github.com/zendframework/zend-servicemanager/pull/272

### CODE REVIEW

**Comment by samsonasik** [comment]
@sasezaki please rebase against latest develop branch

**Comment by sasezaki** [comment]
I give up rebase includes zendframework/zend-servicemanager#272, and would to Reopening of setup.
(Adding baseline & modify it,  to be clear change reason )

**Comment by sasezaki** [comment]
according this vote, 
https://github.com/laminas/technical-steering-committee/blob/main/meetings/minutes/2020-08-03-TSC-Minutes.md#vote-on-static-analysis-tooling
I close this PR .
>

**Comment by Ocramius** [comment]
@sasezaki I think the fixes are valid: just need to swap out the CI tool

File: src/AbstractFactory/ReflectionBasedAbstractFactory.php [hunk 1]
```diff
@@ -10,6 +10,7 @@
 
 use Laminas\ServiceManager\Exception\ServiceNotFoundException;
 use Laminas\ServiceManager\Factory\AbstractFactoryInterface;
+use Laminas\Stdlib\DispatchableInterface;
```
Comment (sasezaki) [inline src/AbstractFactory/ReflectionBasedAbstractFactory.php:13]: This is a mistake for
```
    /**
     * {@inheritDoc}
     *
     * @return DispatchableInterface
     */
    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
```

Not returned DispatchableInterface, just returned object  - requestedName instance

**Pull request closed without merging**